widget/menu.go

```diff
@@ -167,9 +170,33 @@ func (m *Menu) Move(pos fyne.Position) {
 //
 // Implements: fyne.Widget
 func (m *Menu) Refresh() {
+	if m.needsResize() {
+		m.Resize(m.MinSize())
+		return
+	}
+
 	widget.RefreshWidget(m)
 }
 
+func (m *Menu) needsResize() bool {
+	containedCheck := m.containsCheck
+	m.containsCheck = m.getContainsCheck()
+	return containedCheck != m.containsCheck
+}
+
+func (m *Menu) getContainsCheck() bool {
+	for _, item := range m.Items {
+		mi, ok := item.(*menuItem)
+		if !ok {
+			continue
+		}
+		if mi.Item.HasCheckmark {
+			return true
+		}
+	}
```

Comment:
I think this might be cleaner. Haven't tested it, but I think it should work.
```suggestion
	for _, item := range m.Items {
		if mi, ok := item.(*menuItem); ok && mi.Item.HasCheckmark {
			return true
		}
	}
```